pkg/tsdb/loki/api.go

```diff
@@ -156,13 +157,9 @@ func (api *LokiAPI) DataQuery(ctx context.Context, query lokiQuery) (*loghttp.Qu
 		return nil, makeLokiError(resp.Body)
 	}
 
-	var response loghttp.QueryResponse
-	err = jsoniter.NewDecoder(resp.Body).Decode(&response)
-	if err != nil {
-		return nil, err
-	}
-
-	return &response, nil
+	iter := jsoniter.Parse(jsoniter.ConfigDefault, resp.Body, 1024)
```

Comment:
@ryantxu i attempted to skim through the source of `jsoniter` to see how the buffer is used, but wasn't able to quickly confirm the behavior. do you happen to know if there could be any issues with parsing log lines over 1024 bytes?

Reply:
@toddtreece good question, i've just copied it from somewhere else, probably from 
 https://github.com/grafana/grafana/blob/main/pkg/util/converter/prom_test.go#L35 ... i can switch it to whatever is the "right way".

Reply:
@gabor thoughts on adding a test with a log line over 1024 chars as a quick check? (assuming that loki has log lines that long)

Reply:
hmm... it feels a little that we are testing an internal detail, will try to come up with something 👍 .
btw. i did a quick test where i changed `1024` to `1` and it still works fine (probably a lot worse performance of course)

Reply:
@toddtreece btw. if this is a buffer, then i'd assume you control the memory-used vs. performance tradeoff with it, but it should always work.